source/agora/node/Node.d

```diff
@@ -462,6 +464,32 @@ public class Node : API
             node_config.key_pair);
     }
 
+    /***************************************************************************
+
+        Returns an instance of a Nominator.
+
+        Test-suites can inject a badly-behaved nominator in order to
+        simulate byzantine nodes.
+
+        Params:
+            key_pair = the key pair of the node
+            ledger = Ledger instance
+            taskman = the task manager
+            quorum_peers = the network clients of the validator peers
+            quorum_set = the SCP quorum set
+
+        Returns:
+            the enrollment manager
+
+    ***************************************************************************/
+
+    protected Nominator getNominator (KeyPair key_pair, Ledger ledger,
```

Comment:
I'm not sure how I feel about this. We are adding more and more of those functions. Dependency injection is good, but it looks like we're injecting quite a lot.

Reply:
I agree (I originally wrote this) and we need a better way of simulating byzantine nodes. Probably the best way is to not instantiate Node, but our own separate Node implementing the API. I'll sketch something out in the coming days.